sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample2_ExecuteSparkStatement.md

```diff
@@ -0,0 +1,200 @@
+};
+
+SparkSession sessionCreated = client.CreateSparkSession(request);
+
+// Waiting session creation completion
+sessionCreated = PollSparkSession(client, sessionCreated);
+```
+
+## Retrieve a Spark Session
+
+To retrieve an existing session call `GetSparkSession`, passing in the session ID.
+
+```C# Snippet:GetSparkSession
+SparkSession session = client.GetSparkSession(sessionCreated.Id);
+Debug.WriteLine($"Session is returned with name {session.Name} and state {session.State}");
+```
+
+## Creating Spark statements
+
+To create statements within a session call `CreateSparkStatement`, passing in both the statements details in a `SparkStatementOptions` along with the ID of the session.
+
+To wait for the statement's completion a `PollSparkStatement` call is required. This support code will be detailed below, and it hoped to be temporary. 
+
+```C# Snippet:CreateSparkStatement
+SparkStatementOptions sparkStatementRequest = new SparkStatementOptions
+{
+    Kind = SparkStatementLanguageType.Spark,
+    Code = @"print(""Hello world\n"")"
+};
+SparkStatement statementCreated = client.CreateSparkStatement(sessionCreated.Id, sparkStatementRequest);
+
+// Wait operation completion
+statementCreated = PollSparkStatement(client, sessionCreated.Id, statementCreated);
+```
+
+## Retrieve a statement
+
+To retrieve an existing statement call `GetSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:GetSparkStatement
+SparkStatement statement = client.GetSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is returned with id {statement.Id} and state {statement.State}");
+```
+
+## Cancel a statement
+
+To cancel a submitted statement call `CancelSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:CancelSparkStatement
+SparkStatementCancellationResult cancellationResult = client.CancelSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is cancelled with message {cancellationResult.Msg}");
+```
+
+## Cancel a session
+
+To cancel the entire Spark session  call `CancelSparkSession`, passing in the session ID.
+
+```C# Snippet:CancelSparkSession
+Response operation = client.CancelSparkSession(sessionCreated.Id);
+```
+
+## Support Code
+
+Today the following support code is needed to poll the status of submitted Spark statements and sessions.
+
+It is hoped to be temporary.
+
+```C# Snippet:TemporarySparkSupportCode
+ private const string Error = "error";
+ private const string Dead = "dead";
+ private const string Success = "success";
+ private const string Killed = "killed";
+ private const string Idle = "idle";
+
+public static List<string> SessionSubmissionFinalStates = new List<string>
+ {
+     Idle,
+     Error,
+     Dead,
+     Success,
+     Killed
+ };
+
+ public static SparkSession PollSparkSession(
+     SparkSessionClient client,
+     SparkSession session,
+     IList<string> livyReadyStates = null)
```

Comment:
I'm not sure what the `livy` prefix for `livyReadyStates` means.... is that important within this context?